data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/CsaProfileRemedialActionsCreator.java

```diff
@@ -113,17 +110,21 @@ private void createAndAddRemedialActions() {
                             )
                             .collect(Collectors.toList());
 
-                    addOnContingencyStateUsageRules(networkActionAdder, faraoContingenciesIds, randomCombinationConstraintKind);
+                    addOnContingencyStateUsageRules(remedialActionAdder, faraoContingenciesIds, randomCombinationConstraintKind);
                 } else { // no contingency linked to RA --> on instant usage rule
                     String kind = parentRemedialActionPropertyBag.get(CsaProfileConstants.RA_KIND);
                     if (kind.equals(CsaProfileConstants.RemedialActionKind.PREVENTIVE.toString())) {
-                        networkActionAdder.newOnInstantUsageRule().withUsageMethod(UsageMethod.AVAILABLE).withInstant(Instant.PREVENTIVE).add();
+                        remedialActionAdder.newOnInstantUsageRule().withUsageMethod(UsageMethod.AVAILABLE).withInstant(Instant.PREVENTIVE).add();
                     } else {
-                        networkActionAdder.newOnInstantUsageRule().withUsageMethod(UsageMethod.AVAILABLE).withInstant(Instant.CURATIVE).add();
+                        remedialActionAdder.newOnInstantUsageRule().withUsageMethod(UsageMethod.AVAILABLE).withInstant(Instant.CURATIVE).add();
                     }
                 }
 
-                networkActionAdder.add();
+                if (remedialActionAdder instanceof NetworkActionAdder) {
+                    ((NetworkActionAdder) remedialActionAdder).add();
+                } else {
+                    ((PstRangeActionAdder) remedialActionAdder).add();
+                }
```

Comment:
use the enum here to make it more elegant
but i think we should make the add() API easier to use if you don't care about the exact type of returned object (maybe add a RemedialActionAdder.addGeneric() method?)